tests/component/podcasts/test.httpRequests.spec.js

```diff
@@ -0,0 +1,83 @@
+const supertest = require('supertest')
+const { mockIllegalFieldsPodcast, mockPodcast, mockNewPodcast, mockExtraFieldsPodcast } = require('./mockPodcast.js')
+
+const app = require('../../../app')
+
+jest.mock('../../../models/fileModel', () => require('./mockFileModel.js'))
+
+describe('Podcast component test', () => {
+  describe('Add podcast tests', () => {
+    it('should return 200 when podcast is added with all valid fields', async () => {
+      await supertest(app)
+        .post('/podcast/new')
+        .send(mockNewPodcast)
+        .expect(200)
+    })
+    it('should return 400 when trying to add podcast with illegal fields', async () => {
+      await supertest(app)
+        .post('/podcast/new')
+        .send(mockIllegalFieldsPodcast)
+        .expect(400)
+    })
+    it('should return 400 when trying to add podcast extra fields', async () => {
+      await supertest(app)
+        .post('/podcast/new')
+        .send(mockExtraFieldsPodcast)
+        .expect(400)
+    })
+  })
+  describe('Get podcast tests', () => {
+    it('should return 200 when podcast is fetched with existing ID', async () => {
+      await supertest(app)
+        .get('/podcast/1132')
+        .expect(200, mockPodcast)
+    })
+    it('should return 400 when podcast request is not a number', async () => {
+      await supertest(app)
+        .get('/podcast/string')
+        .expect(400)
+    })
+    it('should return 404 when podcast is not in DB', async () => {
+      await supertest(app)
+        .get('/podcast/0')
+        .expect(404)
+    })
+  })
+  describe('Update podcast tests', () => {
+    it('should return 200 when podcast is updated with valid fields', async () => {
+      await supertest(app)
+        .put('/podcast/1132')
+        .send(mockNewPodcast)
+        .expect(200)
+    })
+    it('should return 400 when trying to update podcast with illegal fields', async () => {
+      await supertest(app)
+        .put('/podcast/1132')
+        .send(mockIllegalFieldsPodcast)
+        .expect(400)
+    })
+    it('should return 400 when trying to update podcast with non existing fields', async () => {
+      await supertest(app)
+        .put('/podcast/1132')
+        .send(mockExtraFieldsPodcast)
+        .expect(400)
+    })
+    it('should return 404 when trying to update podcast that does not exist', async () => {
+      await supertest(app)
+        .put('/podcast/0')
+        .expect(404)
+    })
+  })
```

Comment:
it should also return 400 if a bad data was passed, no? for example if in field numberOfEpisodes we are passing a string. this test is missing